### PR TITLE
[veomni] feat: add veomni qwen3-30b and fix ep

### PR DESCRIPTION
**Pull request by wangshuyang31**
### What does this PR do?

>  Add veomni qwen3-30b and qwen3-1.7b npu script 
>  Fix ep bug in transformer_impl.py

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/verl-project/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: examples/ascend_extras/grpo_trainer/run_qwen3-30b_veomni_npu.sh‎ [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+export CKPT_DIR="./ckpt30b"
+
+TRAIN_FILE=dapo-math-17k.parquet
+TEST_FILE=aime-2024.parquet
+max_prompt_length=$((1024 * 2))
+max_response_length=$((1024 * 8))
+rollout_max_num_seqs=$((128))
+
+
+python3 -m verl.trainer.main_ppo \
```
Comment (wuxibin89) [inline examples/ascend_extras/grpo_trainer/run_qwen3-30b_veomni_npu.sh‎:126]: Move this to `examples/grpo_trainer` since it's common to GPU/NPU.

File: examples/ascend_extras/grpo_trainer/run_qwen3_1_7b_npu.sh‎ [hunk 2]
```diff
@@ -0,0 +1,64 @@
+set -x
+ENGINE=${1:-vllm}
+
+FSDP_SIZE=${FSDP_SIZE:-4}
+SP_SIZE=${SP_SIZE:-2}
+EP_SIZE=${EP_SIZE:-1}
+NUM_GPUS=${NUM_GPUS:-16}
+
+TRAIN_FILE=dapo-math-17k.parquet
```
Comment (wuxibin89) [inline examples/ascend_extras/grpo_trainer/run_qwen3_1_7b_npu.sh‎]: Remove this script, we don't keep script for every model.

File: examples/ascend_extras/grpo_trainer/run_qwen3-30b_veomni_npu.sh‎ (hunk 1 above)
Comment (wangshuyang31) [reply examples/ascend_extras/grpo_trainer/run_qwen3-30b_veomni_npu.sh‎:126]: ok, thanks

Comment (wuxibin89) [reply examples/ascend_extras/grpo_trainer/run_qwen3-30b_veomni_npu.sh‎:126]: Script naming rule: `run_<model>_<train-backend>.sh`

File: examples/grpo_trainer/run_qwen3_30b_veomni.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
+set -x
```
Comment (wucong25) [inline examples/grpo_trainer/run_qwen3_30b_veomni.sh:6]: npu后缀需要去掉，现在脚本命名标准为模型_模型大小_训练后端.sh，npu与gpu脚本合并，脚本名不需要体现npu

File: examples/grpo_trainer/run_qwen3-30b_veomni_npu.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
+set -x
+ENGINE=${1:-vllm}
+
+export TASK_QUEUE_ENABLE=1
+export HCCL_OP_EXPANSION_MODE="AIV"
```
Comment (wucong25) [inline examples/grpo_trainer/run_qwen3-30b_veomni_npu.sh]: 这些npu的环境变量放在device判断为npu中，脚本的参数格式参考下其他脚本

File: examples/grpo_trainer/run_qwen3_30b_veomni.sh [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+ENGINE=${1:-vllm}
+DEVICE=${DEVICE:-$(python3 -c 'import torch_npu' 2>/dev/null && echo npu || echo gpu)}
```
Comment (wucong25) [inline examples/grpo_trainer/run_qwen3_30b_veomni.sh:8]: 根据这个DEVICE来判断是否是npu

File: examples/grpo_trainer/run_qwen3_30b_veomni.sh [hunk 6]
```diff
@@ -0,0 +1,88 @@
+set -x
```
Comment (wucong25) [inline examples/grpo_trainer/run_qwen3_30b_veomni.sh:6]: 脚本最上面注明一下算法、模型、训练后端、支持的硬件类型和推理后端，如：#!/usr/bin/env bash
# GRPO | Qwen3-4B | FSDP training | NVIDIA GPUs or Ascend NPUs
#
# INFER_BACKEND controls rollout backend: vllm

**Review by wucong25 — APPROVED** [review]

**Review by wucong25 — APPROVED** [review]
